extensions/amp-share-tracking/0.1/amp-share-tracking.js

```diff
@@ -60,20 +63,13 @@ export class AmpShareTracking extends AMP.BaseElement {
 
   /**
    * Get the incoming share-tracking fragment from the viewer
-   * @return {!Promise<string|undefined>}
+   * @return {!Promise<string>}
    * @private
    */
   getIncomingFragment_() {
     return viewerFor(this.win).getFragment().then(fragment => {
-      if (!fragment || fragment.indexOf('.') != 0) {
-        return;
-      }
-      const endIndex = fragment.indexOf('&');
-      if (endIndex != -1) {
-        return fragment.substr(1, (endIndex - 1));
-      } else {
-        return fragment.substr(1);
-      }
+      const match = fragment.match(/\.([^&]*)/);
+      return match && match[1];
```

Comment:
This can return `null`. We can change to `match ? match[1] : ''` to always return a string.